src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_shazzrah.cpp

```diff
@@ -36,62 +36,80 @@ class boss_shazzrah : public CreatureScript
 
     struct boss_shazzrahAI : public BossAI
     {
-        boss_shazzrahAI(Creature* creature) : BossAI(creature, BOSS_SHAZZRAH) { }
+        boss_shazzrahAI(Creature* creature) : BossAI(creature, BOSS_SHAZZRAH)
+        {
+        }
 
-        void EnterCombat(Unit* target) override
+        void EnterCombat(Unit* /*target*/) override
         {
-            BossAI::EnterCombat(target);
-            events.ScheduleEvent(EVENT_ARCANE_EXPLOSION, 6000);
-            events.ScheduleEvent(EVENT_SHAZZRAH_CURSE, 10000);
-            events.ScheduleEvent(EVENT_MAGIC_GROUNDING, 24000);
-            events.ScheduleEvent(EVENT_COUNTERSPELL, 15000);
-            events.ScheduleEvent(EVENT_SHAZZRAH_GATE, 45000);
+            _EnterCombat();
+            events.ScheduleEvent(EVENT_ARCANE_EXPLOSION, urand(2000, 4000));
+            events.ScheduleEvent(EVENT_SHAZZRAH_CURSE, urand(7000, 11000));
+            events.ScheduleEvent(EVENT_MAGIC_GROUNDING, urand(14000, 19000));
+            events.ScheduleEvent(EVENT_COUNTERSPELL, urand(9000, 10000));
+            events.ScheduleEvent(EVENT_SHAZZRAH_GATE, 30000);
         }
 
         void UpdateAI(uint32 diff) override
         {
             if (!UpdateVictim())
+            {
                 return;
+            }
 
             events.Update(diff);
 
             if (me->HasUnitState(UNIT_STATE_CASTING))
+            {
                 return;
+            }
 
-            while (uint32 eventId = events.ExecuteEvent())
+            while (uint32 const eventId = events.ExecuteEvent())
             {
                 switch (eventId)
                 {
                     case EVENT_ARCANE_EXPLOSION:
+                    {
                         DoCastVictim(SPELL_ARCANE_EXPLOSION);
-                        events.ScheduleEvent(EVENT_ARCANE_EXPLOSION, urand(4000, 7000));
+                        events.RepeatEvent(urand(4000, 5000));
                         break;
-                    // Triggered subsequent to using "Gate of Shazzrah".
+                    }
+                        // Triggered subsequent to using "Gate of Shazzrah".
                     case EVENT_ARCANE_EXPLOSION_TRIGGERED:
+                    {
                         DoCastVictim(SPELL_ARCANE_EXPLOSION);
                         break;
+                    }
                     case EVENT_SHAZZRAH_CURSE:
+                    {
                         if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true, -SPELL_SHAZZRAH_CURSE))
```

Comment:
Does the negated spell enum mean that it will find a target that doesn't have that aura applied?

Reply:
Yes.